libdnf5/repo/repo_downloader.cpp

```diff
@@ -498,8 +517,35 @@ void RepoDownloader::common_handle_setup(LibrepoHandle & h) {
 
         // download the rest metadata added by 3rd parties
         for (auto & item : optional_metadata) {
-            dlist.push_back(item.c_str());
+            // the appstream metadata is a "virtual" type, the list
+            // of the types is read from the repomd file
+            if (item.compare(libdnf5::METADATA_TYPE_APPSTREAM) != 0)
+                dlist.push_back(item.c_str());
         }
+#ifdef WITH_APPSTREAM
```

Comment:
Hmm, we can probably drop this conditional in the end as it is now in a separate plugin. There is not anyway anything dependent on the appstream library.